### PR TITLE
Adding 10th Gen Y/U support

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
Signed-off-by: BrentHoltsclaw <brent.holtsclaw@intel.com>

### CODE REVIEW

File: chipsec/chipset.py [hunk 1]
```diff
@@ -120,238 +121,257 @@ class RegisterType:
...
+Chipset_Dictionary[0x3E32].append({'name' : 'CoffeeLake',     'id' : CHIPSET_ID_CFL , 'code' : CHIPSET_CODE_CFL,  'longname' : 'Desktop 8th Generation Core Processor (CoffeeLake Server 8 Cores)' })
 
 # 8th Generation Core Processor Family (Whiskey Lake)
-0x3E34 : {'name': 'Whiskey Lake',    'id' : CHIPSET_ID_WHL , 'code' : CHIPSET_CODE_WHL,  'longname' : 'Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores)' },
+Chipset_Dictionary[0x3E34].append({'name': 'Whiskey Lake',    'id' : CHIPSET_ID_WHL , 'code' : CHIPSET_CODE_WHL,  'longname' : 'Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores)', 'detection_value' : '806EA' })
+
+# 10th Generation 14nm (Comet Lake)
+Chipset_Dictionary[0x3E35].append({'name' : 'CometLake', 'id' : CHIPSET_ID_CML , 'code' : 'CML', 'longname' : 'CometLake v1 U2 Core'})
+Chipset_Dictionary[0x3E34].append({'name' : 'CometLake', 'id' : CHIPSET_ID_CML , 'code' : 'CML', 'longname' : 'CometLake v1 U4 Core', 'detection_value' : '806EC'})
+Chipset_Dictionary[0x3E33].append({'name' : 'CometLake', 'id' : CHIPSET_ID_CML , 'code' : 'CML', 'longname' : 'CometLake v1 U6 Core', 'detection_value' : '806EC'})
```
Comment (martingalvan) [inline chipsec/chipset.py]: This seems to be the same DID as Desktop 8th Generation Core Processor (CoffeeLake Server 4).

File: chipsec/chipset.py [hunk 2]
```diff
@@ -120,238 +121,257 @@ class RegisterType:
...
+Chipset_Dictionary[0x3ECA].append({'name' : 'CoffeeLake',     'id' : CHIPSET_ID_CFL , 'code' : CHIPSET_CODE_CFL,  'longname' : 'Desktop 8th Generation Core Processor (CoffeeLake Server 6 Cores)' })
+Chipset_Dictionary[0x3E32].append({'name' : 'CoffeeLake',     'id' : CHIPSET_ID_CFL , 'code' : CHIPSET_CODE_CFL,  'longname' : 'Desktop 8th Generation Core Processor (CoffeeLake Server 8 Cores)' })
 
 # 8th Generation Core Processor Family (Whiskey Lake)
-0x3E34 : {'name': 'Whiskey Lake',    'id' : CHIPSET_ID_WHL , 'code' : CHIPSET_CODE_WHL,  'longname' : 'Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores)' },
+Chipset_Dictionary[0x3E34].append({'name': 'Whiskey Lake',    'id' : CHIPSET_ID_WHL , 'code' : CHIPSET_CODE_WHL,  'longname' : 'Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores)', 'detection_value' : '806EA' })
+
+# 10th Generation 14nm (Comet Lake)
+Chipset_Dictionary[0x3E35].append({'name' : 'CometLake', 'id' : CHIPSET_ID_CML , 'code' : 'CML', 'longname' : 'CometLake v1 U2 Core'})
+Chipset_Dictionary[0x3E34].append({'name' : 'CometLake', 'id' : CHIPSET_ID_CML , 'code' : 'CML', 'longname' : 'CometLake v1 U4 Core', 'detection_value' : '806EC'})
```
Comment (martingalvan) [inline chipsec/chipset.py]: This seems to be the same DID as Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores).

File: chipsec/chipset.py (hunk 1 above)
Comment (martingalvan) [reply chipsec/chipset.py]: I just re-checked the CFL and CML datasheets and effectively, the DIDs repeat.

**Comment by BrentHoltsclaw** [comment]
@martingalvan Yes the DID repeat for some of the processors as you indicated.  To address this, duplicate DID support #739 needs to be checked in.  The pull request has been updated to incorporate those changes.  In the cases of duplicate DID, it falls back to another check based upon cpuid.

**Pull request merged**